src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java

```diff
@@ -0,0 +1,147 @@
+      .compose(this::fetchRequests)
+      .compose(this::findServicePointNames)
+      .compose(this::findCallNumbers)
+      .onSuccess(this::buildNewRequests)
+      .compose(this::updateRequests)
+      .onSuccess(r -> log.info("{} processing finished successfully", batch))
+      .recover(t -> handleError(batch, t));
+  }
+
+  @Override
+  RequestSearchMigrationContext buildContext(JsonObject request) {
+    return new RequestSearchMigrationContext(request);
+  }
+
+  private Future<Batch<RequestSearchMigrationContext>> findServicePointNames(
+    Batch<RequestSearchMigrationContext> batch) {
+
+    Set<String> servicePointIds = batch.getRequestMigrationContexts()
+      .stream()
+      .map(RequestSearchMigrationContext::getPickupServicePointId)
+      .filter(Objects::nonNull)
+      .collect(toSet());
+
+    if (servicePointIds.isEmpty()) {
+      return succeededFuture(batch);
+    }
+
+    return okapiClient.get(SERVICE_POINT_URL, servicePointIds, "servicepoints", ServicePoint.class)
+      .onSuccess(servicePoints -> saveServicePointNames(batch, servicePoints))
+      .map(batch);
+  }
+
+  private static void saveServicePointNames(Batch<RequestSearchMigrationContext> batch,
+    Collection<ServicePoint> servicePoints) {
+
+    Map<String, String> servicePointIdToName = servicePoints.stream()
+      .collect(toMap(ServicePoint::getId, ServicePoint::getName));
+
+    batch.getRequestMigrationContexts().forEach(ctx ->
+      ctx.setPickupServicePointName(servicePointIdToName.get(ctx.getPickupServicePointId())));
+  }
+
+  private Future<Batch<RequestSearchMigrationContext>> findCallNumbers(
+    Batch<RequestSearchMigrationContext> batch) {
+
+    Set<String> itemIds = batch.getRequestMigrationContexts()
+      .stream()
+      .map(RequestSearchMigrationContext::getItemId)
+      .filter(Objects::nonNull)
+      .collect(toSet());
+
+    if (itemIds.isEmpty()) {
+      return succeededFuture(batch);
+    }
+
+    return okapiClient.get(ITEMS_STORAGE_URL, itemIds, "items", Item.class)
+      .onSuccess(items -> saveCallNumbers(batch, items))
+      .map(batch);
+  }
+
+  private static void saveCallNumbers(Batch<RequestSearchMigrationContext> batch,
+    Collection<Item> items) {
+
+    Map<String, JsonObject> itemIdToCallNumberComponents = items.stream()
+      .collect(toMap(Item::getId, Item::getEffectiveCallNumberComponents));
+
+    Map<String, String> itemIdToShelvingOrder = items.stream()
+      .collect(toMap(Item::getId, Item::getEffectiveShelvingOrder));
+
+    batch.getRequestMigrationContexts().forEach(ctx -> {
+      ctx.setCallNumberComponents(itemIdToCallNumberComponents.get(ctx.getItemId()));
+      ctx.setShelvingOrder(itemIdToShelvingOrder.get(ctx.getItemId()));
+    });
+  }
+
+  public void buildNewRequest(RequestSearchMigrationContext context) {
+    final JsonObject migratedRequest = context.getOldRequest().copy();
+
+    write(migratedRequest, "pickupServicePointName", context.getPickupServicePointName());
```

Comment:
shouldn't these fields go to "searchIndex"?

Reply:
Fixed